hazelcast/src/test/java/com/hazelcast/ringbuffer/impl/operations/ReadManyOperationTest.java

```diff
@@ -103,15 +108,17 @@ public void whenOneAfterTail() throws Exception {
         assertEquals(0, response.readCount());
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void whenTooFarAfterTail() throws Exception {
         ringbuffer.add("tail");
 
         ReadManyOperation op = getReadManyOperation(ringbuffer.tailSequence() + 2, 1, 1, null);
         op.setNodeEngine(nodeEngine);
 
         // since there is an item, we don't need to wait
-        op.shouldWait();
+        assertFalse(op.shouldWait());
```

Comment:
Before, `op.shouldWait` threw directly. Now `shouldWait` returns false and `beforeRun` throws. This is an implementation detail, therefore I changed the test to comply the new implementation.